indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -179,3 +183,35 @@ def _get_defaults(self):
         coordinators = {x.principal for x in self.session.acl_entries if x.has_management_role(role='coordinate')}
         return {'managers': managers, 'protection_mode': self.session.protection_mode, 'coordinators': coordinators,
                 'acl': acl}
+
+
+class RHManageSessionBlock(RHManageSessionBase):
+    """Manage block of a session"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.session_block
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageSessionBase._checkParams(self, params)
+        self.session_block = SessionBlock.get_one(request.view_args['block_id'])
+
+    def _process(self):
+        form = MeetingSessionBlockForm(obj=FormDefaults(**self._get_form_defaults()), event=self.event_new)
+        if form.validate_on_submit():
+            session_data = {k[8:]: v for k, v in form.data.iteritems() if k in form.session_fields}
+            block_data = {k[6:]: v for k, v in form.data.iteritems() if k in form.block_fields}
+            update_session(self.session, session_data)
+            update_session_block(self.session_block, block_data)
+            tpl = get_template_module('events/timetable/display/_session_block.html')
```

Comment:
Agreed. This is an issue. We need to get the right template, which is not always easy. I'd really drop in-place replacement for now.